Content.Server/Damage/Systems/DamageUserOnTriggerSystem.cs

```diff
@@ -0,0 +1,52 @@
+using Content.Server.Damage.Components;
+using Content.Server.Explosion.EntitySystems;
+using Content.Shared.Damage;
+using Content.Shared.StepTrigger;
+using Content.Shared.StepTrigger.Systems;
+
+namespace Content.Server.Damage.Systems;
+
+// System for damage that occurs on specific trigger, towards the user..
+public sealed class DamageUserOnTriggerSystem : EntitySystem
+{
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<DamageUserOnTriggerComponent, TriggerEvent>(OnTrigger);
+    }
+
+    private void OnTrigger(EntityUid uid, DamageUserOnTriggerComponent component, TriggerEvent args)
+    {
+        if (args.User is null)
+            return;
+
+        args.Handled |= OnDamageTrigger(uid, args.User.Value, component);
+    }
+
+    private bool OnDamageTrigger(EntityUid source, EntityUid target, DamageUserOnTriggerComponent? component = null)
+    {
+        if (!Resolve(source, ref component))
+        {
+            return false;
+        }
+
+        var damage = new DamageSpecifier(component.Damage);
+        var ev = new BeforeDamageUserOnTriggerEvent(damage, target);
+        RaiseLocalEvent(source, ev, true);
```

Comment:
Should this be broadcast?

Reply:
this is oldcode, git fucky with the diff but ill change it